com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs

```diff
@@ -770,30 +770,60 @@ void DoBokehDepthOfField(CommandBuffer cmd, int source, int destination, Rect pi
         #endregion
 
         #region Motion Blur
-
-        void DoMotionBlur(Camera camera, CommandBuffer cmd, int source, int destination)
+#if ENABLE_VR && ENABLE_XR_MODULE
+        // Hold the stereo matrices to avoid allocating arrays every frame
+        internal static readonly Matrix4x4[] viewProjMatrixStereo = new Matrix4x4[2];
+#endif
+        void DoMotionBlur(CameraData cameraData, CommandBuffer cmd, int source, int destination)
         {
             var material = m_Materials.cameraMotionBlur;
 
-            // This is needed because Blit will reset viewproj matrices to identity and UniversalRP currently
-            // relies on SetupCameraProperties instead of handling its own matrices.
-            // TODO: We need get rid of SetupCameraProperties and setup camera matrices in Universal
-            var proj = camera.nonJitteredProjectionMatrix;
-            var view = camera.worldToCameraMatrix;
-            var viewProj = proj * view;
-
-            material.SetMatrix("_ViewProjM", viewProj);
+#if ENABLE_VR && ENABLE_XR_MODULE
+            if (cameraData.xr.enabled && cameraData.xr.singlePassEnabled)
+            {
+                var viewProj0 = GL.GetGPUProjectionMatrix(cameraData.GetProjectionMatrix(0), true) * cameraData.GetViewMatrix(0);
+                var viewProj1 = GL.GetGPUProjectionMatrix(cameraData.GetProjectionMatrix(1), true) * cameraData.GetViewMatrix(1);
+                if (m_ResetHistory)
+                {
+                    viewProjMatrixStereo[0] = viewProj0;
+                    viewProjMatrixStereo[1] = viewProj1;
+                    material.SetMatrixArray("_PrevViewProjMStereo", viewProjMatrixStereo);
+                }
+                else
+                    material.SetMatrixArray("_PrevViewProjMStereo", m_PrevViewProjM);
 
-            if (m_ResetHistory)
-                material.SetMatrix("_PrevViewProjM", viewProj);
+                m_PrevViewProjM[0] = viewProj0;
+                m_PrevViewProjM[1] = viewProj1;
+            }
             else
-                material.SetMatrix("_PrevViewProjM", m_PrevViewProjM);
+#endif
+            {
+                int prevViewProjMIdx = 0;
+#if ENABLE_VR && ENABLE_XR_MODULE
+                if (cameraData.xr.enabled)
+                    prevViewProjMIdx = cameraData.xr.multipassId;
+#endif
+                // This is needed because Blit will reset viewproj matrices to identity and UniversalRP currently
+                // relies on SetupCameraProperties instead of handling its own matrices.
+                // TODO: We need get rid of SetupCameraProperties and setup camera matrices in Universal
```

Comment:
Is this todo tracked somewhere?

Reply:
Hi @pbbastian 
I am not sure about the plan around this todo comment. 
It showed up as changes because of indentation but honestly these lines are not changed.
And I think we should keep this todo comment so that the owner of URP postFX can clear them up later.

Reply:
ohh okay, I see! Never mind this comment then :)